lightning/src/ln/msgs.rs

```diff
@@ -825,8 +825,33 @@ pub trait RoutingMessageHandler : Send + Sync {
 	/// immediately higher (as defined by <PublicKey as Ord>::cmp) than starting_point.
 	/// If None is provided for starting_point, we start at the first node.
 	fn get_next_node_announcements(&self, starting_point: Option<&PublicKey>, batch_amount: u8) -> Vec<NodeAnnouncement>;
-	/// Returns whether a full sync should be requested from a peer.
+	/// Returns whether a full sync should be requested from a peer. The result is
+	/// used to set the initial_routing_sync feature bit when sending an Init message.
+	/// This feature bit is not used by some implementations in favor of using gossip_queries.
 	fn should_request_full_sync(&self, node_id: &PublicKey) -> bool;
+	/// Initiates routing gossip sync by querying a peer to discover channels
+	/// and their associated routing gossip messages. This method will use a
+	/// sync strategy defined by the implementor.
+	fn sync_routing_table(&self, their_node_id: &PublicKey, init: &Init);
+	/// Handles the reply of a query we initiated to learn about channels
+	/// for a given range of blocks. We can expect to receive one or more
+	/// replies to a single query.
+	fn handle_reply_channel_range(&self, their_node_id: &PublicKey, msg: ReplyChannelRange) -> Result<(), LightningError>;
+	/// Handles the reply of a query we initiated asking for routing gossip
+	/// messages for a list of channels. We should receive this message when
+	/// a node has completed its best effort to send us the pertaining routing
+	/// gossip messages.
+	fn handle_reply_short_channel_ids_end(&self, their_node_id: &PublicKey, msg: ReplyShortChannelIdsEnd) -> Result<(), LightningError>;
+	/// Handles when a peer asks us to send a list of short_channel_ids
+	/// for the requested range of blocks. There are potential DoS vectors when
+	/// handling inbound queries. Handling requests with first_blocknum very far
+	/// away may trigger repeated disk I/O if the NetworkGraph is not fully in-memory.
+	fn handle_query_channel_range(&self, their_node_id: &PublicKey, msg: QueryChannelRange) -> Result<(), LightningError>;
+	/// Handles when a peer asks us to send routing gossip messages for a
+	/// list of short_channel_ids. There are potential DoS vectors when handling
+	/// inbound queries. Handling requests with first_blocknum very far away may
+	/// trigger repeated disk I/O if the NetworkGraph is not fully in-memory.
```

Comment:
Alternatively, you can gather potential DoS vectors in "Implementor DoS Warnings" section around `RoutingMessageHandler` trait doc, with trade-offs if graph isn't in memory. I'm fine being in trait function docs, your call if you want to remove reference to `NetworkGraph`.